predicate/parser_test.go

```diff
@@ -0,0 +1,184 @@
+package predicate
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/influxdata/influxdb"
+	influxtesting "github.com/influxdata/influxdb/testing"
+	"github.com/influxdata/influxql"
+)
+
+func TestParseNode(t *testing.T) {
```

Comment:
It's not clear to me why we're adding all this support and test coverage for expressions that we would parse and then reject.